### PR TITLE
Change FileDownloader to http client

### PR DESCRIPTION
**Pull request by yiguolei**

### CODE REVIEW

File: be/src/olap/task/engine_batch_load_task.cpp [hunk 1]
```diff
@@ -113,45 +113,31 @@ AgentStatus EngineBatchLoadTask::_init() {
...
     }
 
     // Empty remote_path
-    if (status == DORIS_SUCCESS && !_push_req.__isset.http_file_path) {
+    if (!_push_req.__isset.http_file_path) {
         _is_init = true;
         return status;
     }
-
+
```
Comment (kangpinghuang) [inline be/src/olap/task/engine_batch_load_task.cpp]: extra empty

File: be/src/olap/task/engine_batch_load_task.cpp [hunk 2]
```diff
@@ -218,96 +172,94 @@ void EngineBatchLoadTask::_get_file_name_from_path(const string& file_path, stri
...
-
-        // Download file from hdfs
-        for (uint32_t i = 0; i < MAX_RETRY; ++i) {
+        bool is_timeout = false;
+        auto download_cb = [this, estimate_time_out, file_size, &is_timeout] (HttpClient* client) {
             // Check timeout and set timeout
-            time_t now = time(nullptr);
-
-            if (_push_req.timeout > 0) {
+            time_t now = time(NULL);
```
Comment (kangpinghuang) [inline be/src/olap/task/engine_batch_load_task.cpp]:
```suggestion
            time_t now = time(nullptr);
```

**Review by chaoyli — APPROVED** [review]

File: be/src/olap/task/engine_batch_load_task.cpp [hunk 3]
```diff
@@ -218,96 +172,94 @@ void EngineBatchLoadTask::_get_file_name_from_path(const string& file_path, stri
...
+            if (_push_req.timeout > 0 && _push_req.timeout < now) {
+                // return status to break this callback
                 VLOG(3) << "check time out. time_out:" << _push_req.timeout
-                        << ", now:" << now;
-                if (_push_req.timeout < now) {
-                    LOG(WARNING) << "push time out";
-                    status = DORIS_PUSH_TIME_OUT;
-                    break;
-                }
+                    << ", now:" << now;
```
Comment (kangpinghuang) [inline be/src/olap/task/engine_batch_load_task.cpp]: change VLOG(3) to LOG(WARNING)

File: be/src/olap/task/engine_batch_load_task.cpp [hunk 4]
```diff
@@ -218,96 +172,94 @@ void EngineBatchLoadTask::_get_file_name_from_path(const string& file_path, stri
...
     if (status == DORIS_SUCCESS) {
         // Load delta file
-        int64_t duration_ns = 0;
-        OLAPStatus push_status = OLAP_SUCCESS;
-        {
-            SCOPED_RAW_TIMER(&duration_ns);
-            push_status = _push(_push_req, _tablet_infos);
-        }
-        LOG(INFO) << "Push finish, cost time: " << duration_ns/1000 << " us";
+        time_t push_begin = time(NULL);
```
Comment (kangpinghuang) [inline be/src/olap/task/engine_batch_load_task.cpp]:
```suggestion
        time_t push_begin = time(nullptr);
```

File: be/src/olap/task/engine_batch_load_task.cpp [hunk 5]
```diff
@@ -218,96 +172,94 @@ void EngineBatchLoadTask::_get_file_name_from_path(const string& file_path, stri
...
+        OLAPStatus push_status = _push(_push_req, _tablet_infos);
+        time_t push_finish = time(NULL);
```
Comment (kangpinghuang) [inline be/src/olap/task/engine_batch_load_task.cpp]:
```suggestion
        time_t push_finish = time(nullptr);
```

File: be/src/olap/task/engine_batch_load_task.cpp [hunk 6]
```diff
@@ -218,96 +172,94 @@ void EngineBatchLoadTask::_get_file_name_from_path(const string& file_path, stri
...
+        LOG(INFO) << "Push finish, cost time: " << (push_finish - push_begin);
```
Comment (kangpinghuang) [inline be/src/olap/task/engine_batch_load_task.cpp]: add time unit

File: be/src/olap/task/engine_batch_load_task.cpp [hunk 7]
```diff
@@ -316,11 +268,9 @@ AgentStatus EngineBatchLoadTask::_process() {
     }
 
     // Delete download file
-    boost::filesystem::path download_file_path(_downloader_param.local_file_path);
-    if (boost::filesystem::exists(download_file_path)) {
-        if (remove(_downloader_param.local_file_path.c_str()) == -1) {
-            OLAP_LOG_WARNING("can not remove file: %s",
-                             _downloader_param.local_file_path.c_str());
+    if (boost::filesystem::exists(_local_file_path)) {
+        if (remove(_local_file_path.c_str()) == -1) {
```
Comment (kangpinghuang) [inline be/src/olap/task/engine_batch_load_task.cpp]:
```suggestion
        if (remove(_local_file_path) == -1) {
```

**Review by kangpinghuang — CHANGES_REQUESTED** [review]

**Pull request merged**